lib/src/analyzer/options/options.dart

```diff
@@ -94,10 +99,25 @@ void _applyOptionsToConfig(
   );
 }
 
+/// Configuration provided in a client project `analysis_options.yaml` under the
+/// `custom_linter` key.
 class RuleConfig {
-  RuleConfig(this.url);
+  /// Creates a new rule configuration with the given properties.
+  RuleConfig({this.documentationUrl});
+
+  /// Whether or not the rules in this package are enabled.
   bool enabled = true;
-  String? url;
+
+  /// {@template documentation_url}
+  /// Optional path to a markdown file describing the rule. The markdown file
+  /// must contain a heading with the same name as the rule id.
```

Comment:
```suggestion
  /// should contain a heading with the same name as the rule id.
```
Technically if the heading does not exist, the web page will just stay scrolled at the top, so not strictly required.